File: BST/2. Search a given key in BST/Java/src/Iterative.java
public class Iterative {

    public static void searchElem(Node root, int elem) {
        if (root == null) {
            System.out.print("Root is empty");
            return;
        }

        Node curr = root;
        while ((curr != null) && (curr.value != elem)) {
            if (elem < curr.value) {
                curr = curr.left;
            }
            else {
                curr = curr.right;
            }
        }

        if (curr == null) {
            System.out.print("Element not found\n");
            return;
        }
        else if (curr.parent == null) {
            System.out.print("Position: Root Node\n");
            return;
        }
        else if (elem < curr.parent.value) {
            System.out.println("Position: Left of Node with key " + curr.parent.value);
            return;
        }
        else  {
            System.out.println("Position: Right of Node with key " + curr.parent.value);
            return;
        }        
    }

    public static void main(String args[]) {
        Node root = null;

        int[] keys = {15, 10, 20, 8, 12, 16, 25, 30};
        for (int i = 0; i < keys.length; i++) {
            root = root.insert(root, keys[i]);
        }

        // root.ascending(root);
        // System.out.print("\n");

        searchElem(root, 20);
    }
}

File: BST/2. Search a given key in BST/python/iterative.py
# from binarytree import Node

class Node:
    left = None
    right = None
    parent = None
    def __init__(self,value):
        self.value = value

def insert(root, key):
    if root is None:
        return Node(key)

    if key < root.value:
        root.left = insert(root.left, key)
        root.left.parent = root 
    else:
        root.right = insert(root.right, key)
        root.right.parent = root 
 
    return root

def ascending(root):
    if root is None:
        return
    ascending(root.left)
    print(root.value, end=' ')
    ascending(root.right)

def searchElem(root,elem):
    if root is None:
        print("Root is empty")
        return

    curr = root

    while((curr) and (curr.value is not elem)):
        if (elem < curr.value):
            curr = curr.left
        else:
            curr = curr.right
    
    if curr is None:
        print("Element not found")
        return
    elif (curr.parent is None):
        print("Position: Root Node")
        return
    elif (elem < curr.parent.value):
        print("Position: Left of Node with key", curr.parent.value)
        return
    else:
        print("Position: Right of Node with key", curr.parent.value)
        return
    
keys = [15, 10, 20, 8, 12, 16, 25,30]

root = None
for key in keys:
    root = insert(root,key)

searchElem(root,30)